packages/wonder-blocks-form/src/components/labeled-text-field.js

```diff
@@ -53,25 +97,42 @@ export default class LabeledTextField extends React.Component<Props, State> {
         });
     };
 
+    handleOnChange: (newValue: string) => mixed = (newValue) => {
+        this.setState({
+            value: newValue,
+        });
+    };
+
     render(): React.Node {
-        const {disabled} = this.props;
+        const {id, type, label, description, disabled} = this.props;
+
         return (
-            <FieldHeading
-                field={
-                    <TextField
-                        id="tf-1"
-                        value="Value"
-                        placeholder="Placeholder"
-                        disabled={disabled}
-                        onChange={() => {}}
-                        onFocus={this.handleOnFocus}
-                        onBlur={this.handleOnBlur}
+            <IDProvider id={id} scope="field">
+                {(uniqueId) => (
+                    <FieldHeading
+                        id={uniqueId}
+                        field={
+                            <TextField
+                                id={`${uniqueId}-field`}
+                                aria-describedby={`${uniqueId}-error`}
+                                aria-invalid={
+                                    this.state.error ? "true" : "false"
+                                }
+                                type={type}
+                                value={this.state.value}
+                                placeholder="Placeholder"
+                                disabled={disabled}
+                                onChange={this.handleOnChange}
+                                onFocus={this.handleOnFocus}
+                                onBlur={this.handleOnBlur}
+                            />
+                        }
+                        label={label}
+                        description={description}
+                        error="Error"
```

Comment:
Is the plan to add the error `prop` as part of https://khanacademy.atlassian.net/browse/WB-1068?

Reply:
I don't necessarily plan to add `error` as a prop, but keep it as a state. The `error` state will be used to pass the error message `onValidation` returns to `FieldHeading`'s error label. I will indeed be making use of this on that next ticket.

Reply:
Oh good point! I totally forgot about that part 😅. That makes total sense.